### PR TITLE
Added configuration proxy

### PR DESCRIPTION
**Pull request by areksolek**
Added configuration proxy

### CODE REVIEW

File: languages/en-US/Settings/_Base.json [hunk 1]
```diff
@@ -385,6 +386,15 @@
...
     "LBL_ADVANCED_RECORD_NUMBERING": "Picklist sequence settings",
     "LBL_PREFIX": "Prefix",
+    "LBL_CONFIG_PROXY": "Proxy",
+    "LBL_CONFIG_PROXY_DESCRIPTION": "Proxy",
+    "LBL_PROXY_CONNECTION": "Connection",
+    "LBL_PROXY_PROTOCOL": "Protocol",
+    "LBL_PROXY_HOST": "Host",
+    "LBL_PROXY_PORT": "Port",
+    "LBL_PROXY_LOGIN": "Login",
+    "LBL_PROXY_PASSWORD": "Password",
```
Comment (rskrzypczak) [inline languages/en-US/Settings/_Base.json]: move to languages/en-US/Settings/Proxy.json

File: layouts/basic/modules/Settings/Proxy/ConfigProxyDetail.tpl [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{*<!-- {[The file is published on the basis of YetiForce Public License 3.0 that can be found in the following directory: licenses/LicenseEN.txt or yetiforce.com]} -->*}
+{strip}
+
```
Comment (rskrzypczak) [inline layouts/basic/modules/Settings/Proxy/ConfigProxyDetail.tpl]: `<!-- tpl ..`

File: layouts/basic/modules/Settings/Proxy/ConfigProxyEdit.tpl [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+/*********************************************************************************
+** The contents of this file are subject to the vtiger CRM Public License Version 1.0
+* ("License"); You may not use this file except in compliance with the License
+* The Original Code is:  vtiger CRM Open Source
+* The Initial Developer of the Original Code is vtiger.
+* Portions created by vtiger are Copyright (C) vtiger.
+* All Rights Reserved.
+* Contributor(s): YetiForce Sp. z o.o.
+********************************************************************************/
+-->*}
```
Comment (rskrzypczak) [inline layouts/basic/modules/Settings/Proxy/ConfigProxyEdit.tpl]: wrong license and rename file and use `<!-- tpl..`

File: modules/Settings/Proxy/actions/ConfigProxySaveAjax.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
+<?php
+/* +**********************************************************************************
+ * The contents of this file are subject to the vtiger CRM Public License Version 1.1
+ * ("License"); You may not use this file except in compliance with the License
+ * The Original Code is:  vtiger CRM Open Source
+ * The Initial Developer of the Original Code is vtiger.
+ * Portions created by vtiger are Copyright (C) vtiger.
+ * All Rights Reserved.
+ * Contributor(s): YetiForce Sp. z o.o.
+ * ********************************************************************************** */
```
Comment (rskrzypczak) [inline modules/Settings/Proxy/actions/ConfigProxySaveAjax.php]: wrong license and rename file

File: layouts/basic/modules/Settings/Proxy/ConfigProxyEdit.tpl [hunk 5]
```diff
@@ -0,0 +1,48 @@
+{*<!-- {[The file is published on the basis of YetiForce Public License 3.0 that can be found in the following directory: licenses/LicenseEN.txt or yetiforce.com]} -->*}
+{strip}
+	<div class="tpl-Settings-Vtiger-ConfigProxyEdit">
```
Comment (rskrzypczak) [inline layouts/basic/modules/Settings/Proxy/ConfigProxyEdit.tpl]:
```suggestion
	<div class="tpl-Settings-Proxy-Edit">
```

File: layouts/basic/modules/Settings/Proxy/ConfigProxyEdit.tpl [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+					<tr class="blockHeader">
+						<th colspan="2"
+							class="{$WIDTHTYPE}">{\App\Language::translate('LBL_PROXY_CONFIGRATION', $QUALIFIED_MODULE)}</th>
+					</tr>
+					</thead>
+					<tbody>
+					{foreach key=FIELD_NAME item=FIELD_LABEL from=$MODEL->listFields}
+						{assign var="FIELD_MODEL" value=$MODEL->getFieldInstanceByName($FIELD_NAME)->set('fieldvalue',$MODEL->get($FIELD_NAME))}
+						<tr>
+							<td width="30%" class="{$WIDTHTYPE} textAlignRight">
```
Comment (rskrzypczak) [inline layouts/basic/modules/Settings/Proxy/ConfigProxyEdit.tpl]: deprecated: textAlignRight

File: modules/Settings/Proxy/actions/ConfigProxySaveAjax.php [hunk 7]
```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Settings proxy config view file.
+ *
+ * @copyright YetiForce Sp. z o.o
+ * @license YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author Arkadiusz Sołek <a.solek@yetiforce.com>
+ */
+
+class Settings_Proxy_ConfigProxySaveAjax_Action extends Settings_Vtiger_Basic_Action
```
Comment (rskrzypczak) [inline modules/Settings/Proxy/actions/ConfigProxySaveAjax.php]:
```suggestion
class Settings_Proxy_SaveAjax_Action extends Settings_Vtiger_Basic_Action
```
and add @package

File: modules/Settings/Proxy/models/ConfigModule.php [hunk 8]
```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Settings proxy config module model.
+ *
+ * @copyright YetiForce Sp. z o.o
+ * @license YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author Arkadiusz Sołek <a.solek@yetiforce.com>
+ */
+
+class Settings_Proxy_ConfigModule_Model extends Settings_Vtiger_Module_Model
```
Comment (rskrzypczak) [inline modules/Settings/Proxy/models/ConfigModule.php]:
```suggestion
class Settings_Proxy_Module_Model extends Settings_Vtiger_Module_Model
```

File: modules/Settings/Proxy/views/ConfigProxyEdit.php [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+ * Settings proxy config module model.
+ *
+ * @copyright YetiForce Sp. z o.o
+ * @license YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author Arkadiusz So�ek <a.solek@yetiforce.com>
+ */
+
+class Settings_Proxy_ConfigProxyEdit_View extends Settings_Vtiger_Index_View
+{
+	public function process(App\Request $request)
```
Comment (rskrzypczak) [inline modules/Settings/Proxy/views/ConfigProxyEdit.php]: dec

File: modules/Settings/Proxy/views/ConfigProxyEdit.php [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+		$viewer->view('ConfigProxyEdit.tpl', $qualifiedName);
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function getFooterScripts(App\Request $request)
+	{
+		return array_merge(parent::getFooterScripts($request), $this->checkAndConvertJsScripts([
+			"modules.Settings.Proxy.resources.ConfigEditor",
```
Comment (rskrzypczak) [inline modules/Settings/Proxy/views/ConfigProxyEdit.php]: ?

File: modules/Settings/Proxy/actions/ConfigProxySaveAjax.php [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+		$response = new Vtiger_Response();
+		$qualifiedModuleName = $request->getModule(false);
+		$moduleModel = Settings_Proxy_ConfigModule_Model::getInstance();
+		try {
+			$configFile = new \App\ConfigFile('security');
+			foreach (array_keys($moduleModel->listFields) as $fieldName) {
+				if ($request->has($fieldName)) {
+					if ($fieldName === 'proxyConnection') {
+						$proxyValue = false;
+						if ($request->getRaw($fieldName) === '1') {
```
Comment (rskrzypczak) [inline modules/Settings/Proxy/actions/ConfigProxySaveAjax.php]: getBoolen

File: modules/Settings/Proxy/actions/ConfigProxySaveAjax.php [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+							$proxyValue = true;
+						}
+					} else {
+							$proxyValue = $request->getRaw($fieldName);
```
Comment (rskrzypczak) [inline modules/Settings/Proxy/actions/ConfigProxySaveAjax.php]: changed getRaw

File: modules/Settings/Proxy/models/ConfigModule.php [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+	}
+
+	/**
+	 * Function determines fields available in edition view.
+	 *
+	 * @param string $name
+	 *
+	 * @return \Settings_Vtiger_Field_Model
+	 */
+	public function getFieldInstanceByName($name)
```
Comment (rskrzypczak) [inline modules/Settings/Proxy/models/ConfigModule.php]:
```suggestion
	public function getFieldInstanceByName(string $name)
```

File: modules/Settings/Proxy/views/ConfigProxyEdit.php [hunk 14]
```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * Settings proxy config module model.
+ *
+ * @copyright YetiForce Sp. z o.o
+ * @license YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author Arkadiusz So�ek <a.solek@yetiforce.com>
```
Comment (rskrzypczak) [inline modules/Settings/Proxy/views/ConfigProxyEdit.php]: coding

File: modules/Settings/Proxy/actions/SaveAjax.php [hunk 15]
```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Settings proxy save ajax action.
+ *
+ * @package   Settings.Actions
```
Comment (rskrzypczak) [inline modules/Settings/Proxy/actions/SaveAjax.php]:
```suggestion
 * @package   Settings.Action
```

File: modules/Settings/Proxy/models/Module.php [hunk 16]
```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Settings proxy config module model.
+ *
+ * @package   Settings.Module
```
Comment (rskrzypczak) [inline modules/Settings/Proxy/models/Module.php]:
```suggestion
 * @package   Settings.Model
```

File: modules/Settings/Proxy/models/Module.php [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+		foreach ($this->listFields as $fieldName => $fieldData) {
+			$value = \App\Config::security($fieldName);
+			$this->set($fieldName, $value);
+		}
+	}
+
+	/**
+	 * Function to get the instance of Config module model.
+	 *
+	 * @param mixed $name
```
Comment (rskrzypczak) [inline modules/Settings/Proxy/models/Module.php]:
```suggestion
	 * @param string $name
```

File: modules/Settings/Proxy/models/Module.php [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+	 *
+	 * @throws \ReflectionException
+	 *
+	 * @return \Settings_Proxy_Module_Model|\Settings_Vtiger_Module_Model
```
Comment (rskrzypczak) [inline modules/Settings/Proxy/models/Module.php]:
```suggestion
	 * @return self
```
or @return Settings_Proxy_Module_Model

File: public_html/layouts/basic/modules/Settings/Proxy/resources/Index.js [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+		registerEvents: function () {
+				this.registerEditViewEvents();
+		}
+	}
+);
+
+jQuery(document).ready(function (e) {
+	var tacInstance = new Settings_Proxy_Index_Js();
+	tacInstance.registerEvents();
+});
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Settings/Proxy/resources/Index.js]: remove this

File: public_html/layouts/basic/modules/Settings/Proxy/resources/Index.js [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+		 * function to register the events in editView
+		 */
+		registerEditViewEvents: function () {
+			let thisInstance = this;
+			let form = $('#ConfigProxyForm');
+			App.Fields.Picklist.showSelect2ElementView(form.find('select.select2'), {
+				dropdownCss: { 'z-index': 0 }
+			});
+			let fieldsChange = form.find(
+				"*[name='proxyConnection'], *[name='proxyProtocol'], *[name='proxyHost'], *[name='proxyPort'], *[name='proxyLogin'], *[name='proxyPassword']"
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Settings/Proxy/resources/Index.js]: ?

File: public_html/layouts/basic/modules/Settings/Proxy/resources/Edit.js [hunk 21]
```diff
@@ -33,42 +33,41 @@ jQuery.Class(
...
 			});
-			let fieldsChange = form.find(
-				"*[name='proxyConnection'], *[name='proxyProtocol'], *[name='proxyHost'], *[name='proxyPort'], *[name='proxyLogin'], *[name='proxyPassword']"
-			  );
+			let fieldsChange = form.find('.js-proxy-field');
 			form.validationEngine(app.validationEngineOptions);
-			fieldsChange.each(function() {
+			fieldsChange.find('input, select').each(function () {
 				let element = $(this);
 				element.on('change', function (e) {
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Settings/Proxy/resources/Edit.js]:
```suggestion
			fieldsChange.find('input, select').on('change', function (e) {
```

File: modules/Settings/Proxy/actions/SaveAjax.php [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+	public function process(App\Request $request)
+	{
+		$response = new Vtiger_Response();
+		$qualifiedModuleName = $request->getModule(false);
+		$moduleModel = Settings_Proxy_Module_Model::getInstance();
+		try {
+			$configFile = new \App\ConfigFile('security');
+			foreach (array_keys($moduleModel->listFields) as $fieldName) {
+				if ($request->has($fieldName)) {
+					if ($request->getBoolean($fieldName)) {
```
Comment (rskrzypczak) [inline modules/Settings/Proxy/actions/SaveAjax.php:29]: ?

File: layouts/basic/modules/Settings/Proxy/Edit.tpl [hunk 23]
```diff
@@ -0,0 +1,52 @@
...
+			<thead>
+				<tr class="blockHeader">
+					<th colspan="2" class="{$WIDTHTYPE}">{\App\Language::translate('LBL_PROXY_CONFIGRATION',
+						$QUALIFIED_MODULE)}
+					</th>
+				</tr>
+			</thead>
+			<tbody>
+				{foreach key=FIELD_NAME item=FIELD_LABEL from=$MODULE_MODEL->listFields}
+				{assign var="FIELD_MODEL"
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Settings/Proxy/Edit.tpl:31]: ?

File: modules/Settings/Proxy/actions/SaveAjax.php [hunk 24]
```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Settings proxy save ajax action.
+ *
+ * @package   Settings.Action
+ *
+ * @copyright YetiForce Sp. z o.o
+ * @license YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author Arkadiusz Sołek <a.solek@yetiforce.com>
+ */
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Proxy/actions/SaveAjax.php:10]: doc

File: modules/Settings/Proxy/models/Module.php [hunk 25]
```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * Settings proxy config module model.
+ *
+ * @package   Settings.Model
+ *
+ * @copyright YetiForce Sp. z o.o
+ * @license YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author Arkadiusz Sołek <a.solek@yetiforce.com>
+ */
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Proxy/models/Module.php:10]: doc

File: modules/Settings/Proxy/models/Module.php [hunk 26]
```diff
@@ -0,0 +1,97 @@
...
+class Settings_Proxy_Module_Model extends Settings_Vtiger_Module_Model
+{
+	/**
+	 * Load proxy module list fields.
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Proxy/models/Module.php:14]: ?

File: modules/Settings/Proxy/models/Module.php [hunk 27]
```diff
@@ -0,0 +1,97 @@
...
+	}
+
+	/**
+	 * Function determines fields available in edition view.
+	 *
+	 * @param string $name
+	 *
+	 * @return \Settings_Vtiger_Field_Model
+	 */
+	public function getFieldInstanceByName(string $name)
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Proxy/models/Module.php:61]: ?

File: modules/Settings/Proxy/views/Edit.php [hunk 28]
```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Settings proxy edit view.
+ *
+ * @package   Settings.View
+ *
+ * @copyright YetiForce Sp. z o.o
+ * @license YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author Arkadiusz Sołek <a.solek@yetiforce.com>
+ */
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Proxy/views/Edit.php:10]: doc

File: modules/Settings/Proxy/views/Edit.php [hunk 29]
```diff
@@ -0,0 +1,36 @@
...
+class Settings_Proxy_Edit_View extends Settings_Vtiger_Index_View
+{
+	/**
+	 * Process action.
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Proxy/views/Edit.php:14]: ?

File: modules/Settings/Proxy/views/Edit.php [hunk 30]
```diff
@@ -0,0 +1,36 @@
...
+		$viewer->view('Edit.tpl', $qualifiedName);
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function getFooterScripts(App\Request $request)
+	{
+		return array_merge(parent::getFooterScripts($request), $this->checkAndConvertJsScripts([
+			'modules.Settings.Proxy.resources.Edit',
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/Proxy/views/Edit.php:33]: ?

**Review by mariuszkrzaczkowski — CHANGES_REQUESTED** [review]

**Pull request closed without merging**